addons/survey/models/survey_question.py

```diff
@@ -127,7 +139,10 @@ def default_get(self, fields):
         ('validation_length', 'CHECK (validation_length_min <= validation_length_max)', 'Max length cannot be smaller than min length!'),
         ('validation_float', 'CHECK (validation_min_float_value <= validation_max_float_value)', 'Max value cannot be smaller than min value!'),
         ('validation_date', 'CHECK (validation_min_date <= validation_max_date)', 'Max date cannot be smaller than min date!'),
-        ('validation_datetime', 'CHECK (validation_min_datetime <= validation_max_datetime)','Max datetime cannot be smaller than min datetime!')
+        ('validation_datetime', 'CHECK (validation_min_datetime <= validation_max_datetime)', 'Max datetime cannot be smaller than min datetime!'),
+        ('positive_answer_score', 'CHECK (answer_score >= 0)', 'An answer score for a non-multiple choice question cannot be negative!'),
+        ('scored_questions_have_answers', "CHECK ((is_scored_question!=True) OR (question_type ='numerical_box') OR (date_answer is not null) OR (datetime_answer is not null))",
+            'All "Is a scored question = True" questions need an answer')
```

Comment:
Smart. I split check by question types `date` and `datetime`. Unfortunately there's no way to ensure that users type in a `numerical_box_answer` since it will default to 0 even without user input. Also updating the check this way means we do not have to clean previously entered answers when switching `question_type`, so I deleted the corresponding compute logic.